modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -316,4 +436,30 @@ public CouchbaseContainer withGamesIMSample(boolean gamesIMSample) {
         this.gamesIMSample = gamesIMSample;
         return self();
     }
+
+    @Getter
+    @RequiredArgsConstructor
+    protected enum CouchbasePort {
+        REST("rest_port", 8091, true),
+        CAPI("capi_port", 8092, false),
+        QUERY("query_port", 8093, false),
+        FTS("fts_http_port", 8094, false),
+        CBAS("cbas_http_port", 8095, false),
+        EVENTING("eventing_http_port", 8096, false),
+        MEMCACHED_SSL("memcached_ssl_port", 11207, false),
+        MEMCACHED("memcached_port", 11210, false),
+        REST_SSL("ssl_rest_port", 18091, true),
+        CAPI_SSL("ssl_capi_port", 18092, false),
+        QUERY_SSL("ssl_query_port", 18093, false),
+        FTS_SSL("fts_ssl_port", 18094, false),
+        CBAS_SSL("cbas_ssl_port", 18095, false),
+        EVENTING_SSL("eventing_ssl_port", 18096, false),
+        ;
+
+        final String name;
+
+        final int originalPort;
+
+        final boolean isBootstrap;
```

Comment:
I would rename to `dynamic`, and also mark MEMCACHED as dynamic (as it was in the previous implementation)

Reply:
renaming is fine.


Problem with MEMCACHED is, that when marking them as dynamic, the sdk does not work:
```
11:26:49.578 WARN  com.couchbase.client.core.endpoint.Endpoint - [localhost:11210][KeyValueEndpoint]: Could not connect to endpoint on reconnect attempt 5, retrying with delay 32 MILLISECONDS: com.couchbase.client.deps.io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:11210
11:26:49.612 WARN  com.couchbase.client.core.endpoint.Endpoint - [localhost:11210][KeyValueEndpoint]: Could not connect to remote socket: Connection refused: localhost/127.0.0.1:11210
```
It uses the right mapped port for bootstrapping but then **discovers** the unmapped port from the cluster and then tries to connect to that port and fails with connection refused.